### PR TITLE
Adaptive load helper mocks

### PR DESCRIPTION
**Pull request by eric846**
- MockNighthawkServiceClient
- MockMetricsEvaluator
- MockAdaptiveLoadSessionSpecProtoHelper

Part 8 of splitting PR #483.

### CODE REVIEW

**Review by oschaaf — DISMISSED** [review]

File: test/mocks/common/mock_nighthawk_service_client.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "nighthawk/common/nighthawk_service_client.h"
+
+#include "gmock/gmock.h"
+
+namespace Nighthawk {
+
+class MockNighthawkServiceClient : public NighthawkServiceClient {
```
Comment (dubious90) [inline test/mocks/common/mock_nighthawk_service_client.h:9]: Same comment here.

**Review by dubious90 — COMMENTED** [review]
Looks good.  Just a couple places where some comments would be helpful.

Thanks for breaking these out.  They're much easier to digest now!

File: test/mocks/adaptive_load/mock_metrics_evaluator.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "nighthawk/adaptive_load/metrics_evaluator.h"
+
+#include "gmock/gmock.h"
+
+namespace Nighthawk {
+
+class MockMetricsEvaluator : public MetricsEvaluator {
```
[the remark this answers is not in the document]
Comment (eric846) [reply test/mocks/adaptive_load/mock_metrics_evaluator.h:20]: Done.

File: test/mocks/adaptive_load/mock_metrics_evaluator.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public:
+  MockMetricsEvaluator();
+
+  MOCK_CONST_METHOD3(EvaluateMetric,
```
[the remark this answers is not in the document]
Comment (eric846) [reply test/mocks/adaptive_load/mock_metrics_evaluator.h:27]: Yep

File: test/mocks/adaptive_load/mock_session_spec_proto_helper.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "nighthawk/adaptive_load/session_spec_proto_helper.h"
+
+#include "gmock/gmock.h"
+
+namespace Nighthawk {
+
+class MockAdaptiveLoadSessionSpecProtoHelper : public AdaptiveLoadSessionSpecProtoHelper {
```
[the remark this answers is not in the document]
Comment (eric846) [reply test/mocks/adaptive_load/mock_session_spec_proto_helper.h:28]: Done

File: test/mocks/common/mock_nighthawk_service_client.h (hunk 1 above)
Comment (eric846) [reply test/mocks/common/mock_nighthawk_service_client.h:19]: Done

File: test/mocks/adaptive_load/mock_metrics_evaluator.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (dubious90) [reply test/mocks/adaptive_load/mock_metrics_evaluator.h:20]: Super helpful.  Thank you

**Review by dubious90 — APPROVED** [review]
Looks great. Thanks for adding the comments

**Pull request merged**